snapcraft/internal/repo/snaps.py

```diff
@@ -109,8 +108,8 @@ def get_current_channel(self):
         if self.installed:
             local_snap_info = self.get_local_snap_info()
             current_channel = local_snap_info['channel']
-            if current_channel == 'stable':
-                current_channel = 'latest/stable'
+            if current_channel in ('stable', 'candidate', 'beta', 'edge'):
+                current_channel = 'latest/{}'.format(current_channel)
```

Comment:
This check won't catch branches, think `build-snaps: go/hotfix-211`. Perhaps instead of a whitelist of risks, just check to see if a track is specified (e.g. searching for `/`)?